pkg/kubelet/qos/policy.go

```diff
@@ -40,18 +42,22 @@ const (
 // multiplied by 10 (barring exceptional cases) + a configurable quantity which is between -1000
 // and 1000. Containers with higher OOM scores are killed if the system runs out of memory.
 // See https://lwn.net/Articles/391222/ for more information.
-func GetContainerOOMScoreAdjust(pod *v1.Pod, container *v1.Container, memoryCapacity int64) int {
+func GetContainerOOMScoreAdjust(pod *v1.Pod, container *v1.Container, memoryCapacity int64) (int, error) {
+	if memoryCapacity <= 0 {
+		return 0, errors.New("memory capacity for the host should not less than or equal to zero")
```

Comment:
Would it make sense to include `memoryCapacity` value into the error message?

Reply:
> Would it make sense to include `memoryCapacity` value into the error message?

OK

Reply:
After digging a while, I found that all the usage of `errors.New()` in kubernetes are input with const string, maybe this is better for writing unit test I think? How about we keep this error here and add a `klog.Errorf` function where the `GetContainerOOMScoreAdjust ` is called and contain more detail info like `memoryCapacity` and pod name and container. @bart0sh

Reply:
> After digging a while, I found that all the usage of errors.New() in kubernetes are input with const string

You can use [fmt.Errorf](https://pkg.go.dev/fmt#Errorf) if you want to format an error.

Reply:
I've already changed as you suggested @bart0sh